### PR TITLE
[smarty] use default to prevent error_log pollution

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Updating to use php 8 and updating composer.json dependencies results in smarty polluting error_log about undefined variables. $some_variable|default will mimic the existing behaviour and have the variable be set to an empty string if undefined. 

#### Testing instructions (if applicable)

1. Checkout PR.
2. Visit Loris everywhere

#### Link(s) to related issue(s)

* Resolves #  (Reference the issue this fixes, if any.)

### CODE REVIEW

**Comment by driusan** [comment]
Why is this blocked?

**Comment by maltheism** [comment]
@driusan not all fixes have been added yet. I used git cherry-pick but I'm missing some. I'll remove blocked tag when they're all added. I'm manually going through loris right now.

**Comment by maltheism** [comment]
@driusan I believe it's ready. I might of missed something but I'm doubtful that happened.

File: src/Middleware/UserPageDecorationMiddleware.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ function ($a, $b) {
 
         $tpl_data['subtest'] = $request->getAttribute("pageclass")->page ?? null;
 
-        $page = $request->getAttribute("pageclass");
+        $page = $request->getAttribute("pageclass") ?? '';
```
Comment (maltheism) [inline src/Middleware/UserPageDecorationMiddleware.php:138]: this made it so I could view other smarty problems. @driusan maybe I should undo it?

File: modules/statistics/php/stats_behavioural.class.inc [hunk 2]
```diff
@@ -245,7 +245,13 @@ class Stats_Behavioural extends \NDB_Form
             $center = $row['CenterID'];
             $vl     = $row['VLabel'];
             $value  = intval($row['val']);
-            $c      =& $this->tpl_data['behaviour']['C' . $center];
+            if (!isset($this->tpl_data['behaviour']['C' . $center])) {
```
Comment (maltheism) [inline modules/statistics/php/stats_behavioural.class.inc:248]: @driusan the fixes in here maybe should be moved to a separate PR? Solves error_pollution and might not be the prettiest code. @laemtl you wouldn't happen to know a better way? I believe you were the one that cleaned up this file.

Comment (laemtl) [reply modules/statistics/php/stats_behavioural.class.inc:248]: What about 
$c =& $this->tpl_data['behaviour']['C' . $center][$vl] ?? [$vl => []];

**Comment by driusan** [comment]
Shouldn't this update the smarty version in composer.json/lock too?

File: modules/statistics/php/stats_behavioural.class.inc (hunk 2 above)
Comment (maltheism) [reply modules/statistics/php/stats_behavioural.class.inc:248]: @laemtl my IDE is giving me the warning: `Only variables can be passed by reference` message. Can you test that for me whenever you have time and check your error_log? The error log gets polluted when visiting the statistics module and clicking on the Behavioural Statistics tab.

Comment (laemtl) [reply modules/statistics/php/stats_behavioural.class.inc:248]: @maltheism This shoul works:
$this->tpl_data['behaviour']['C' . $center][$vl] = $this->tpl_data['behaviour']['C' . $center][$vl] ?? [];
$c =& $this->tpl_data['behaviour']['C' . $center];

**Comment by maltheism** [comment]
@driusan ready

**Comment by maltheism** [comment]
Added Priority High tag since https://github.com/aces/Loris/pull/7157 has it and this PR is needed to be merged for it.
